File: leetcode/2433/solution.py
"""
https://leetcode.com/problems/find-the-original-array-of-prefix-xor/
2433. Find the Original Array of Prefix Xor
"""
from typing import List


class Solution:
    def findArray(self, pref: List[int]) -> List[int]:
        arr = [pref[0]] * len(pref)
        for i in range(1, len(pref)):
            arr[i] = pref[i] ^ pref[i - 1]
        return arr

File: leetcode/2433/solution.go
/*
https://leetcode.com/problems/find-the-original-array-of-prefix-xor/
2433. Find the Original Array of Prefix Xor
*/
package main

func findArray(pref []int) []int {
	arr := make([]int, len(pref))
	arr[0] = pref[0]
	for i := 1; i < len(pref); i++ {
		arr[i] = pref[i] ^ pref[i-1]
	}
	return arr
}
